### PR TITLE
toolchain: fix bracket-depth gcc build error when using clang

### PR DESCRIPTION
**Pull request by adriweb**
And align some things here and there...

### CODE REVIEW

**Comment by Vogtinator** [comment]
That kind of alignment only makes editing harder...

**Comment by adriweb** [comment]
well maybe but it's much easier to _read_ and since we basically never edit those lines...

File: ndless-sdk/toolchain/build_toolchain.sh [hunk 1]
```diff
@@ -39,38 +39,42 @@ mkdir -p "$PREFIX"
...
-	wget -c ftp://sourceware.org/pub/newlib/$NEWLIB.tar.gz -O download/$NEWLIB.tar.gz && tar xvzf download/$NEWLIB.tar.gz && \
+	wget -c ftp://ftp.gnu.org/gnu/gcc/$GCC/$GCC.tar.bz2       -O download/$GCC.tar.bz2      && tar xvjf download/$GCC.tar.bz2 && \
+	wget -c ftp://ftp.gnu.org/gnu/gdb/$GDB.tar.xz             -O download/$GDB.tar.xz       && tar xvJf download/$GDB.tar.xz && \
+	wget -c ftp://sourceware.org/pub/newlib/$NEWLIB.tar.gz    -O download/$NEWLIB.tar.gz    && tar xvzf download/$NEWLIB.tar.gz && \
 	touch .downloaded
 	if [ $? -ne 0 ]; then
 		echo "Download failed!"
 		exit 1
 	fi
 fi
```
Comment (Vogtinator) [inline ndless-sdk/toolchain/build_toolchain.sh]: Please always initialize variables in bash. Everything should still run fine with ```set +u```

File: ndless-sdk/toolchain/build_toolchain.sh [hunk 2]
```diff
@@ -39,38 +39,42 @@ mkdir -p "$PREFIX"
...
 
+if [ ! $(gcc -v 2>&1 | grep clang > /dev/null) ]; then
+    EXTRAGCCOPTS='CXXFLAGS="-fbracket-depth=512"'
```
Comment (Vogtinator) [inline ndless-sdk/toolchain/build_toolchain.sh]: I doubt that this works, it should be an option to ./configure instead, maybe using environment variables

File: ndless-sdk/toolchain/build_toolchain.sh (hunk 1 above)
Comment (adriweb) [inline ndless-sdk/toolchain/build_toolchain.sh]: done

File: ndless-sdk/toolchain/build_toolchain.sh [hunk 3]
```diff
@@ -39,38 +39,43 @@ mkdir -p "$PREFIX"
...
+fi
+
 # Section 1: GNU Binutils.
 echo "Building Binutils..."
 [ -f .built_binutils ] || (cd build && rm -rf * && ../$BINUTILS/configure $OPTIONS_BINUTILS && make $PARALLEL all && make install && cd .. && rm -rf build/* && touch .built_binutils) || exit 1;
 
 # Section 2: GCC, step 1.
 echo "Building GCC (step 1)..."
-[ -f .built_gcc_step1 ] || (cd build && rm -rf * && ../$GCC/configure $OPTIONS_GCC && make $PARALLEL all-gcc && make install-gcc && cd .. && rm -rf build/* && touch .built_gcc_step1) || exit 1;
+[ -f .built_gcc_step1 ] || (cd build && rm -rf * && ../$GCC/configure $OPTIONS_GCC && make $PARALLEL $EXTRAGCCOPTS all-gcc && make install-gcc && cd .. && rm -rf build/* && touch .built_gcc_step1) || exit 1;
```
Comment (Vogtinator) [inline ndless-sdk/toolchain/build_toolchain.sh]: Please move to configure as it's also done on https://github.com/busterb/musl-cross-make/blob/fcf2604a5a1c91f2fd590af83c2dc691a6ee49a2/litecross/Makefile

Comment (adriweb) [inline ndless-sdk/toolchain/build_toolchain.sh]: last time I tried that, I don't believe it fixed it, but I can try again...

File: ndless-sdk/toolchain/build_toolchain.sh [hunk 4]
```diff
@@ -30,6 +30,10 @@ OPTIONS_GCC="--target=$TARGET --prefix=$PREFIX --enable-interwork --enable-multi
 OPTIONS_NEWLIB="--target=$TARGET --prefix=$PREFIX --enable-interwork --enable-multilib --with-gnu-as --with-gnu-ld --disable-newlib-may-supply-syscalls --disable-newlib-supplied-syscalls --with-float=soft --disable-werror --disable-nls --enable-newlib-io-float"
 OPTIONS_GDB="--target=$TARGET --prefix=$PREFIX --enable-interwork --enable-multilib --disable-werror --with-python"
 
+if [ ! $(gcc -v 2>&1 | grep clang > /dev/null) ]; then
```
Comment (Vogtinator) [inline ndless-sdk/toolchain/build_toolchain.sh]: You test the output of *nothing* here, that just can't work.

Just do
```if (gcc -v 2>&1 | grep clang > /dev/null); then```

Looks better as well!

Comment (Vogtinator) [inline ndless-sdk/toolchain/build_toolchain.sh]: Also, add a comment why this is needed

**Comment by adriweb** [comment]
Done (I edited online directly, I guess you'll have to squash merge...)

**Comment by adriweb** [comment]
or not :P

**Pull request merged**